Enhanced footer (#525)
kitconcept/volto-light-theme: docs/reference/slots.md

Reviewer: Add toctree entry to `/index.md` for this file.

Author: Done

---
myst:
  html_meta:
    "description": "Volto Light Theme slots definitions"
    "property=og:description": "Volto Light Theme slots definitions"
    "property=og:title": "Slots"
    "keywords": "Volto Light Theme, Slots"
---

# Slots

> Volto Light Theme provides a set of slots that can be used to customize the layout and structure of your pages.

Reviewer: ```suggestion
Volto Light Theme provides a set of slots to customize the layout and structure of your pages.
```

Slots are predefined areas in the page where you can place components.

> Learn more about slots in the [Volto documentation](https://6.docs.plone.org/volto/configuration/slots.html).

Reviewer: Let's make sure I got the syntax correct in the PR preview. Intersphinx should resolve this, as we configured it in `conf.py`.

```suggestion
Learn more about slots in the Volto documentation chapter {doc}`plone:/volto/configuration/slots`.
```

## Usage

You can add a component to a predefined slot by registering it in your add-on's configuration.

```ts
import FooterLogos from '@kitconcept/volto-light-theme/components/Footer/slots/FooterLogos';

config.registerSlotComponent({
  name: 'footerLogos',
  slot: 'preFooter',
  component: FooterLogos,
});
```

The `FooterLogos` component will be rendered in the `preFooter` slot of the page.

> ## `aboveHeader` slot
> The `aboveHeader` slot is a slot that can be used to add content above the header of the page.
> 
> ## `belowHeader` slot
> The `belowHeader` slot is a slot that can be used to add content below the header of the page.
> 
> ## `headerTools` slot
> The `headerTools` slot is a slot that can be used to add content in the header tools (top-most right) area of the page.
> 
> ## `preFooter` slot
> The `preFooter` slot is a slot that can be used to add content before the footer of the page.
> 
> ## `postFooter` slot
> The `postFooter` slot is a slot that can be used to add content after the footer of the page.
> 
> ## `footer` slot
> The `footer` slot is a slot that can be used to add content in the main footer area of the page.
> 
> ## `followUs` slot
> The `followUs` slot is a slot that can be used to add content in the footer "followUs" area of the page.
> 
> ## `footerLinks` slot
> The `footerLinks` slot is a slot that can be used to add content in the footer "links" area of the page.

Reviewer: Let's convert all the headers to a definition list and simplify the text.

```suggestion
`aboveHeader`
:   The `aboveHeader` slot adds content above the header of the page.

`belowHeader`
:   The `belowHeader` slot adds content below the header of the page.

`headerTools`
:   The `headerTools` slot adds content in the header tools at the top-most right area of the page.

`preFooter`
:   The `preFooter` slot adds content before the footer of the page.

`postFooter`
:   The `postFooter` slot adds content after the footer of the page.

`footer`
:   The `footer` slot adds content in the main footer area of the page.

`followUs`
:   The `followUs` slot adds content in the footer "follow us" area of the page.

`footerLinks`
:   The `footerLinks` slot adds content in the footer "links" area of the page.
```
